controllers/apps/cluster/transformer_cluster_component.go

```diff
@@ -986,6 +986,56 @@ func (h *clusterShardingHandler) buildAnnotations(transCtx *clusterTransformCont
 	return annotations
 }
 
+func (h *clusterShardingHandler) buildShardSchedulingPolicy(transCtx *clusterTransformContext,
+	shardingName, compName string, comp *appsv1.Component) {
+	var affinity *corev1.Affinity
+	if comp.Spec.SchedulingPolicy != nil {
+		affinity = comp.Spec.SchedulingPolicy.Affinity // topologySpreadConstraints?
+	}
+	if affinity == nil || (affinity.PodAffinity == nil && affinity.PodAntiAffinity == nil) {
+		h.buildShardPodAntiAffinity(transCtx, shardingName, compName, comp) // fallback
+		return
+	}
+
+	replace := func(terms1 []corev1.PodAffinityTerm, terms2 []corev1.WeightedPodAffinityTerm) bool {
+		found := false
+		for i := range terms1 {
+			for k, v := range terms1[i].LabelSelector.MatchLabels {
```

Comment:
`MatchLabels` is a map, so we don't have to iterate over it to find the specified key.